python/pyarrow/tests/test_array.py

```diff
@@ -2162,6 +2163,101 @@ def test_array_from_numpy_ascii():
     assert arrow_arr.equals(expected)
 
 
+@pytest.mark.nopandas
+def test_interval_array_from_timedelta():
+    data = [
+        None,
+        datetime.timedelta(days=1, seconds=1, microseconds=1,
+                           milliseconds=1, minutes=1, hours=1, weeks=1)]
+
+    # From timedelta (explicit type required)
+    arr = pa.array(data, pa.month_day_nano_interval())
+    assert isinstance(arr, pa.MonthDayNanoIntervalArray)
+    assert arr.type == pa.month_day_nano_interval()
+    expected_list = [
+        None,
+        pa.MonthDayNanoTuple([0, 8,
```

Comment:
Can you test with a non-zero number of months?

Reply:
I might be missing something but it doesn't appear  timedelta has a months or years field, which would be what is required to have a non-zero months fields (otherwise days is sufficient).